### PR TITLE
Not enough information of process state during an OOM

### LINKED ISSUE #7031: Not enough information of process state during an OOM
#### Problem

Not enough information of process state during an OOM.

#### Proposed Solution

OOMs should dump all the threads.

### PR DESCRIPTION
**Pull request by cyberbono3**
#### Problem

Not enough information of process state during an OOM

#### Summary of Changes

Following @sakridge suggestion :
- Add File struct with HashMap that maps current process's name to amount of bytes allocated
- Add write_to_file and from_file methods for serialization and deserialization of File struct.
- Add test test_thread_memory_map() to test this functionality 
test thread_mem_usage::tests::test_thread_memory_map ... ok

Fixes #7031

Need some guidance to proceed further.

### CODE REVIEW

File: core/src/thread_mem_usage.rs [hunk 1]
```diff
@@ -37,3 +42,49 @@ impl Allocatedp {
...
+        let mem = adp.get();
+
+        let current_thread = thread::current();
+        let name = current_thread.name().unwrap();
+
+        let mut map: HashMap<String, u64> = HashMap::new();
+        map.insert(name.to_string(), mem);
+
+        let file = MapFile(map);
+        let path = Path::new("file.txt");
```
Comment (sakridge) [inline core/src/thread_mem_usage.rs]: Can your test put the file into `farf` and also cleanup the file in the case there is an error.

File: core/file.txt [hunk 2]
```diff
@@ -0,0 +1,3 @@
+{
+  "thread_mem_usage::tests::test_thread_memory_map": 128
```
Comment (sakridge) [inline core/file.txt]: dont' want to add this file to git

File: core/src/thread_mem_usage.rs (hunk 1 above)
Comment (cyberbono3) [reply core/src/thread_mem_usage.rs]: it removes a file in a default case. Does it sound good?

**Comment by cyberbono3** [comment]
@sakridge does it look good?any feedback?

**Pull request closed without merging**